src/HealthChecks.Redis/RedisHealthCheck.cs

```diff
@@ -63,6 +63,7 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
             }
             catch (Exception ex)
             {
+                _connections.TryRemove(_redisConnectionString, out _);

```

Comment:
I doubt sbout `TryGetValue/TryAdd` pair above. See #903, #905 . Also note that `ConnectionMultiplexer` dictionary value implements `IDisposable` and should be disposed.

Reply:
I don't get the first part, sorry.
Would you like that, instead of using `TryGetValue/TryAdd/TryRemove` I update it to match the approach from the PR that you have posted?
Related with IDisposable you are right, I'll update my PR with that 😄

Reply:
> Would you like that, instead of using TryGetValue/TryAdd/TryRemove I update it to match the approach from the PR that you have posted?

Yes. In fact that should be done in many places to get rid of race conditions.

Reply:
Thanks for the clarification ❤️ 
I'll update my PR with both changes

Reply:
I'm doing the change and I have faced with a problem using GetOrAdd, it hasn't got an async method. That pushes me to use something like this:
```csharp
var connection = _connections.GetOrAdd(_redisConnectionString, async key =>
{
    return ConnectionMultiplexer.Connect(key);
});                
```

My concern here is about doing sync calls inside the async context of the health check execution. Maybe I create a bigger problem with that than the original race condition with TryAdd/TryGet/TryRemove.
Any thoughts/suggestion @sungam3r ?

Reply:
OK. Let's keep as is for now.